packages/api/lambdas/sf-event-sqs-to-db-records/write-granules.js

```diff
@@ -147,7 +147,26 @@ const generateFileRecord = ({ file, granuleCumulusId }) => ({
 const generateFileRecords = async ({
   files,
   granuleCumulusId,
-}) => files.map((file) => generateFileRecord({ file, granuleCumulusId }));
+}) => {
+  const errors = [];
+
+  const generatedFiles = files.map((file) => {
+    let generatedFile;
+
+    try {
+      generatedFile = generateFileRecord({ file, granuleCumulusId });
+    } catch (error) {
+      errors.push(error);
+    }
```

Comment:
Collecting errors here so that we don't stop on the first one and continue to generate any other valid files.

Reply:
I'd recommend rewriting this entire function using `p-map` like so:

```
return pMap(
   files,
   (file) => generateFileRecord({ file, granuleCumulusId }),
   {
     stopOnError: false,
   }
 )
```

It's a little cleaner and `stopOnError: false` automatically handles the behavior of waiting for all promises to settle then rejecting with an aggregate error for any errors

https://github.com/sindresorhus/p-map#stoponerror

Reply:
This was refactored here: https://github.com/nasa/cumulus/pull/2129/commits/bf8eff5425d57c2c3b944ae7fce6b078ef29e050

We're now using `pMap` for the upserts, not the file generation. Also removed the async/await on the file generation here:

https://github.com/nasa/cumulus/pull/2129/commits/d32bb5a6b24e871241c8f69a20094250453057b1

Shouldn't need it because the translate function isn't async.